core/stateful_precompile_test.go

```diff
@@ -315,19 +314,29 @@ func TestContractNativeMinterRun(t *testing.T) {
 			},
 			suppliedGas: precompile.MintGasCost,
 			readOnly:    false,
-			setupState: func(state *state.StateDB) {
-				precompile.SetContractNativeMinterStatus(state, adminAddr, precompile.AllowListAdmin)
-			},
 			expectedErr: precompile.ErrCannotMint.Error(),
+		},
+		"mint funds from allow address": {
+			caller:         allowAddr,
+			precompileAddr: precompile.ContractNativeMinterAddress,
+			input: func() []byte {
+				input, err := precompile.PackMintInput(allowAddr, common.Big1)
+				if err != nil {
+					panic(err)
+				}
+				return input
+			},
+			suppliedGas: precompile.MintGasCost,
+			readOnly:    false,
+			expectedRes: []byte{},
 			assertState: func(t *testing.T, state *state.StateDB) {
-				res := precompile.GetContractNativeMinterStatus(state, adminAddr)
-				assert.Equal(t, precompile.AllowListAdmin, res)
+				res := precompile.GetContractNativeMinterStatus(state, allowAddr)
+				assert.Equal(t, precompile.AllowListEnabled, res)
 
-				res = precompile.GetContractNativeMinterStatus(state, noRoleAddr)
-				assert.Equal(t, precompile.AllowListAdmin, res)
+				assert.Equal(t, common.Big1, state.GetBalance(allowAddr), "expected increased admin funds")
```

Comment:
```suggestion
				assert.Equal(t, common.Big1, state.GetBalance(allowAddr), "expected increased funds")
```
this is not the admin address